Source/ARTPresenceMap.m

```diff
@@ -150,22 +150,23 @@ - (void)leaveMembersNotPresentInSync {
     }
 }
 
-- (void)reenterLocalMembersMissingFromSync {
-    [_logger debug:__FILE__ line:__LINE__ message:@"%p reentering local members missed from sync (syncSessionId=%lu)", self, (unsigned long)_syncSessionId];
-    NSSet *filteredLocalMembers = [_localMembers filteredSetUsingPredicate:[NSPredicate predicateWithFormat:@"syncSessionId != %lu", (unsigned long)_syncSessionId]];
-    for (ARTPresenceMessage *localMember in filteredLocalMembers) {
+- (void)reset {
+    _members = [NSMutableDictionary dictionary];
+    _localMembers = [NSMutableDictionary dictionary];
+}
+
+- (void)reenterLocalMembers {
+    [_localMembers enumerateKeysAndObjectsUsingBlock:^(NSString * _Nonnull key,
+                                                       ARTPresenceMessage * _Nonnull localMember,
+                                                       BOOL * _Nonnull stop) {
         ARTPresenceMessage *reenter = [localMember copy];
         [self internalRemove:localMember];
```

Comment:
I may be misunderstanding - but why is this removing the member before reentering?

Reply:
Short answer is I don't really know as this was the case before doing this change. Spec change didn't really point out any change in terms of removing or not removing local members. 

But I realized that removing this will not really impact any tests. So I removed it here 9a5f1d8a8babca415e73335b32cb501461b2ec6f . 

However, I'm not sure if that's the correct behaviour and whether this has been coded differently in other SDKs.
Some questions
* Do we have a spec where we have to clear / remove local members before doing reentry? I have quickly searched the spec but couldn't find a corresponding point. but I hope @SimonWoolf could point out one if it exists.
* What if reentry fails? I can see other actions (for example absent members as a result of sync) triggers this `internalRemove` but I am not 100% sure whether the actions in the spec are covering all cases where the local members are cleared when necessary. 

I think this might have been put as a defensive mechanism just in case.

Reply:
>  Do we have a spec where we have to clear / remove local members before doing reentry?

Don't think so. And ably-js certainly doesn't.

> What if reentry fails?

RTP17e covers what happens if reentry fails.

I can't think of a good reason to remove the member from the local map when re-entering. And if the connection disconnects again at the right time so the re-enter fails, you can imagine that that presence member might get lost. So yeah, :+1: to removing that

Reply:
Thanks Simon for your input 👍  In this case @andydunstall are you happy to close this conversation ?